src/Microsoft.ML.TimeSeries/DTRootCauseLocalization.cs

```diff
@@ -0,0 +1,305 @@
+            : base(host, ctx)
+        {
+        }
+
+        // Factory method for SignatureLoadDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // <base>
+            base.SaveColumns(ctx);
+        }
+
+        private protected override IRowMapper MakeRowMapper(DataViewSchema schema) => new Mapper(this, schema);
+
+        private protected override void CheckInputColumn(DataViewSchema inputSchema, int col, int srcCol)
+        {
+            if (!(inputSchema[srcCol].Type is RootCauseLocalizationInputDataViewType))
+                throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", ColumnPairs[col].inputColumnName, "RootCauseLocalizationInputDataViewType", inputSchema[srcCol].Type.ToString());
+        }
+
+        private sealed class Mapper : OneToOneMapperBase
+        {
+            private readonly DTRootCauseLocalizationTransformer _parent;
+
+            public Mapper(DTRootCauseLocalizationTransformer parent, DataViewSchema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+            }
+
+            protected override DataViewSchema.DetachedColumn[] GetOutputColumnsCore()
+            {
+                var result = new DataViewSchema.DetachedColumn[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    InputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].inputColumnName, out int colIndex);
+                    Host.Assert(colIndex >= 0);
+
+                    DataViewType type;
+                    type = new RootCauseDataViewType();
+
+                    result[i] = new DataViewSchema.DetachedColumn(_parent.ColumnPairs[i].outputColumnName, type, null);
+                }
+                return result;
+            }
+
+            protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer)
+            {
+                Contracts.AssertValue(input);
+                Contracts.Assert(0 <= iinfo && iinfo < _parent.ColumnPairs.Length);
+
+                var src = default(RootCauseLocalizationInput);
+                var getSrc = input.GetGetter<RootCauseLocalizationInput>(input.Schema[ColMapNewToOld[iinfo]]);
+
+                disposer =
```

Comment:
>disposer [](start = 16, length = 8)

I think this is not necessary. This is used by components that use resources that are `IDisposable`, for example, `ImageResizer` that uses a `Bitmap` for its computations.

Reply:
Have changed, please help review whether the change is right

Reply:
> > disposer [](start = 16, length = 8)
> 
> I think this is not necessary. This is used by components that use resources that are `IDisposable`, for example, `ImageResizer` that uses a `Bitmap` for its computations.

Have changed, please help review whether the change is right

Reply:
You can also assign `null` instead.

---
In reply to: [399045482](https://github.com/dotnet/machinelearning/pull/4925#discussion_r399045482) [](ancestors = 399045482)

Reply:
> You can also assign `null` instead.
> 
> In reply to: [399045482](https://github.com/dotnet/machinelearning/pull/4925#discussion_r399045482) [](ancestors = 399045482)

Have updated to null